src/vm/analysis/analysis_db.rs

```diff
@@ -88,20 +88,32 @@ impl <'a> AnalysisDatabase <'a> {
     }
 
     pub fn get_public_function_type(&mut self, contract_identifier: &QualifiedContractIdentifier, function_name: &str) -> CheckResult<Option<FunctionType>> {
+        // TODO: this function loads the whole contract to obtain the function type.
+        //         but it doesn't need to -- rather this information can just be 
+        //         stored as its own entry. the analysis cost tracking currently only
+        //         charges based on the function type size.
```

Comment:
But, this shouldn't be asymptotically different, right?

Reply:
It depends on the granularity of the inputs to the asymptotic -- while a function's signature is bound by the contract size, the function signature is usually much smaller.